File: programme/pangram-checker/pangram-checker.go
// Write a programme to check Pangram sentence
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	fmt.Print("Input  : ")
	sc := bufio.NewScanner(os.Stdin) // Scanner can read line with space.

	if sc.Scan() {
		str := strings.ToLower(sc.Text())

		charMap := make(map[string]struct{})

		for i := 0; i < len(str); i++ {
			if 97 <= str[i] && str[i] <= 122 { // 97 == 'a', 122 == 'z' (ascii code)
				charMap[string(str[i])] = struct{}{}
			}
		}

		if len(charMap) == 26 {
			fmt.Println("Output : Pangram Sentence")
		} else {
			fmt.Println("Output : Not a Pangram Sentence")
		}
	}
}

File: programme/pangram-checker/pangram-checker.cs
// Write a programme to check Pangram sentence
using System;
using System.Collections.Generic;

public class PangramChecker
{
    public static void Main(string[] args)
    {
        Console.Write("Input  : ");
        var str = Console.ReadLine();
        if (str == null)
        {
            Console.WriteLine("Input Error");
            return;
        }

        var chars = str.ToCharArray();
        var hs = new HashSet<char>();

        for (int i = 0; i < chars.Length; i++)
        {
            int c = (int) chars[i];

            // 97 == 'a', 122 == 'z' (ascii code)
            if (97 <= c && c <= 122) hs.Add(chars[i]);
        }

        Console
            .WriteLine("Output : {0}Pangram Sentence",
            hs.Count == 26 ? "" : "Not a ");
    }
}
